tests/performance_tests/performance_test_basic.py

```diff
@@ -21,26 +21,144 @@
+        print("Total elapsed: ", self.total_elapsed())
+        print("Total time: ", self.total_time())
+        print("Total latency: ", self.total_latency())
+    def print_range(self):
+        print("Starting block %d ending block %d" % (self.startBlock, self.ceaseBlock))
+
+
+def waitForEmptyBlocks(node):
     emptyBlocks = 0
     while emptyBlocks < emptyBlockGoal:
-        headBlock = node1.getHeadBlockNum()
-        block = node1.processCurlCmd("chain", "get_block_info", f'{{"block_num":{headBlock}}}', silentErrors=False, exitOnError=True)
-        node1.waitForHeadToAdvance()
+        headBlock = node.getHeadBlockNum()
+        block = node.processCurlCmd("chain", "get_block_info", f'{{"block_num":{headBlock}}}', silentErrors=False, exitOnError=True)
+        node.waitForHeadToAdvance()
         if block['transaction_mroot'] == "0000000000000000000000000000000000000000000000000000000000000000":
             emptyBlocks += 1
         else:
             emptyBlocks = 0
+    return node.getHeadBlockNum()
 
-def checkTotalTrx():
-    total = 0
-    f = open("var/lib/node_00/stderr.txt")
-    stringResult = re.findall(r'trxs:\s+\d+', f.read())
-    for value in stringResult:
-        intResult = re.findall(r'\d+', value)
-        total += int(intResult[0])
+def fetchStats(total):
+    i = -1
+    f = open("var/lib/node_01/stderr.txt")
+    trxResult = re.findall(r'trxs:\s+\d+.*cpu:\s+\d+.*', f.read())
+    for value in trxResult:
+        i+=1
+        strResult = re.findall(r'trxs:\s+\d+', value)
+        for str in strResult:
+            intResult = re.findall(r'\d+', str)
+            total.blockLog.append(blockData())
+            total.blockLog[i].transactions = int(intResult[0])
+    i = -1
+    for value in trxResult:
+        i+=1
+        strResult = re.findall(r'cpu:\s+\d+', value)
+        for str in strResult:
+            intResult = re.findall(r'\d+', str)
+            total.blockLog[i].cpu = int(intResult[0])
+    i = -1
+    for value in trxResult:
+        i+=1
+        strResult = re.findall(r'net:\s+\d+', value)
+        for str in strResult:
+            intResult = re.findall(r'\d+', str)
+            total.blockLog[i].net = int(intResult[0])
+    i = -1
+    for value in trxResult:
+        i+=1
+        strResult = re.findall(r'elapsed:\s+\d+', value)
+        for str in strResult:
+            intResult = re.findall(r'\d+', str)
+            total.blockLog[i].elapsed = int(intResult[0])
+    i = -1
+    for value in trxResult:
+        i+=1
+        strResult = re.findall(r'time:\s+\d+', value)
+        for str in strResult:
+            intResult = re.findall(r'\d+', str)
+            total.blockLog[i].time = int(intResult[0])
+    i = -1
+    for value in trxResult:
+        i+=1
+        strResult = re.findall(r'latency:\s+.*\d+', value)
+        for str in strResult:
+            intResult = re.findall(r'-*\d+', str)
+            total.blockLog[i].latency = int(intResult[0])
```

Comment:
This seems like it could all be done in one iteration through all `values` in `trxResult` to avoid the index gymnastics.  Can you get set all `blockData` fields in one go?